src/core/worker.js

```diff
@@ -545,6 +548,15 @@ class WorkerMessageHandler {
           return stream.bytes;
         }
 
+        acroForm = isDict(acroForm) ? acroForm : Dict.empty;
+        const xfa = acroForm.get("XFA") || [];
+        let xfaDatasets = null;
+        for (let i = 0, ii = xfa.length; i < ii; i += 2) {
+          if (xfa[i] === "datasets") {
```

Comment:
I don't think this always works. XFA can not only be an array, but also a stream; see https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf#page=439&zoom=auto,-215,230 and and https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf#page=473&zoom=auto,-215,214.

Reply:
If I understand correctly the specs, then it means that we won't have a reference for the datasets stream and so it'll be a way harder to update it without rewriting all the file...